js/Dot.js

```diff
@@ -1050,80 +1047,160 @@ Track.prototype.width = function(newWidth) {
 }
 
 
-var colorScale = d3.scaleOrdinal(d3.schemeAccent);
+const colorScale = d3.scaleOrdinal(d3.schemeAccent);
 
+// resolveDirection : {strand {'+', '-', other}, degree [-180,180)} -> {degree, degree-180}
+const resolveDirection = (strand, degree) =>
+		R.cond([
+			  [R.equals('+'), R.always(degree)],
+			  [R.equals('-'), R.always(degree-180)],
+			  [R.T, R.always(degree)]
+		  ])(strand);
 
 Track.prototype.draw = function() {
 	this.element.attr("transform", "translate(" + this.state.left + "," + this.state.top + ")");
 
-
 	var selected = this.selected;
 	// Add background or border to track
 	this.element.select("rect.trackBackground")
 		.attr("width", this.state.width)
 		.attr("height", this.state.height)
 
 	this.updateSelected();
-	
+
 	var xOrY = this.side;
 	var scale = this.parent.scales[xOrY];
 
 	var refOrQuery = this.parent.k[xOrY];
-	
+
 	var annotMatches = function(d) {
 		return scale.contains(d[refOrQuery], d[refOrQuery+"_start"]) && scale.contains(d[refOrQuery], d[refOrQuery+"_end"])
 	};
 
 	function scaleAnnot(d) {
-		return {
+		// Include :: Mapping -> Mapping
+		let obj = {
 			start: scale.get(d[refOrQuery], d[refOrQuery + '_start']),
 			end: scale.get(d[refOrQuery], d[refOrQuery + '_end']),
 			seq: d[refOrQuery],
 			hover: d.name + " (" + d[refOrQuery] + ":" + d[refOrQuery + '_start'] + "-" + d[refOrQuery + '_end'] + ")",
 		};
+		return d.strand ? R.merge(obj, { strand: d.strand} ) : obj;
 	}
-	
+
 	var dataToPlot = R.compose(R.map(scaleAnnot), R.filter(annotMatches))(this.data);
 
 	var dataZoomed = zoomFilterSnap(this.parent.scales.zoom.area, this.parent.scales.zoom, xOrY)(dataToPlot);
 
-	var annots = this.element.selectAll(".annot").data(dataZoomed);
-	var newAnnots = annots.enter().append("rect")
-		.attr("class","annot");
+	var _track = this;
 
-	annots.exit().remove();
+	if (_track.arrowToggled != _track.get_styles()["arrowToggled"]) {
```

Comment:
Not sure why this extra part is necessary. Keep this simple by not having a _track.arrowToggled at all, just use it straight from this.styles or using your this.get_styles method to set that symbol variable directly.